orca-core/src/main/java/com/netflix/spinnaker/orca/StageResolver.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca;
+
+import com.netflix.spinnaker.orca.pipeline.ExecutionRunner;
+import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder;
+
+import java.util.Collection;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+public class StageResolver {
+  private final Collection<StageDefinitionBuilder> stageDefinitionBuilders;
+
+  public StageResolver(Collection<StageDefinitionBuilder> stageDefinitionBuilders) {
+    this.stageDefinitionBuilders = stageDefinitionBuilders;
+  }
+
+  public StageDefinitionBuilder getStageDefinitionBuilder(String type, String typeAlias) {
+    return stageDefinitionBuilders
```

Comment:
This is 100% lifted from the `StageDefinitionBuilderFactory` ... idea would be that a `StageDefinitionBuilder` could define 0..* aliases that would also be checked.

These aliases would reflect previous bean names for a particular `StageDefinitionBuilder` if it's ever been renamed.